crates/ra_hir_ty/src/infer.rs

```diff
@@ -363,14 +363,28 @@ impl<'a, D: HirDatabase> InferenceContext<'a, D> {
     }
 
     fn resolve_associated_type(&mut self, inner_ty: Ty, assoc_ty: Option<TypeAliasId>) -> Ty {
+        self.resolve_associated_type_with_params(inner_ty, assoc_ty, &[])
+    }
+
+    fn resolve_associated_type_with_params(
+        &mut self,
+        inner_ty: Ty,
+        assoc_ty: Option<TypeAliasId>,
+        params: &[Ty],
+    ) -> Ty {
         match assoc_ty {
             Some(res_assoc_ty) => {
                 let ty = self.table.new_type_var();
+                let mut builder = Substs::builder(1 + params.len()).push(inner_ty);
```

Comment:
There's a nice helper to ensure the right amount of params is supplied (not sure if the code compiles, but should give the general idea):
```suggestion
                let mut builder = Substs::build_for_def(&self.db, res_assoc_ty).push(inner_ty);
```

Reply:
Oh, nice. I do not know about that helper.